scripts/process-npm-assets.sh

```diff
@@ -0,0 +1,136 @@
+				echo "$USAGE"
+				exit 1
+			fi
+			NODE_MODULES_PATH="$1"
+			shift
+			;;
+		--help)
+			echo "$USAGE"
+			exit 0
+			;;
+		*)
+			echo "Error: Unrecognized option: $1"
+			echo "$USAGE"
+			exit 1
+			;;
+	esac
+done
+
+echo "-------------------------------------------------------------------------"
+echo "Post-processing npm assets...."
+echo "  Working directory          == '$(pwd)'"
+echo "  NODE_MODULES_PATH          == '$NODE_MODULES_PATH'"
+echo "  CSS_VENDOR_PATH            == '$CSS_VENDOR_PATH'"
+echo "  JS_VENDOR_PATH             == '$JS_VENDOR_PATH'"
+echo "  EDX_UI_TOOLKIT_VENDOR_PATH == '$EDX_UI_TOOLKIT_VENDOR_PATH'"
+echo "-------------------------------------------------------------------------"
+
+# Input validation
+if ! [ -d "$NODE_MODULES_PATH" ]; then
+	echo "Error: not a directory: $NODE_MODULES_PATH"
+	exit 1
+fi
+if ! [ -d ./common ]; then
+	echo "Error: not a directory: ./common"
+	echo "Hint: $0 must be run from the root of the edx-platform directory!"
+	exit 1
+fi
+
+# Echo lines back to user.
+set -x
+
+# Create vendor directories.
+mkdir -p "$JS_VENDOR_PATH"
+mkdir -p "$CSS_VENDOR_PATH"
+mkdir -p "$EDX_UI_TOOLKIT_VENDOR_PATH"
+
+# Copy studio-frontend assets into into vendor directory.
+find "$NODE_MODULES_PATH/@edx/studio-frontend/dist" \
+	-type f \! -name \*.css \! -name \*.css.map -print0 | \
+	xargs --null cp --target-directory="$JS_VENDOR_PATH"
+find "$NODE_MODULES_PATH/@edx/studio-frontend/dist" \
+	-type f \( -name \*.css -o -name \*.css.map \) -print0 | \
+	xargs --null cp --target-directory="$CSS_VENDOR_PATH"
+
+# Copy edx-ui-toolkit's JS to into its own special directory.
+# Note: $EDX_UI_TOOLKIT_VENDOR_PATH/js used to be a git-managed symlink.
+#       To avoid confusing behavior for folks who still have the js/ symlink
+#       hanging around in their repo, the safest thing to do is to remove
+#       the target js/ before copying in js/ fresh from the source.
+rm -rf "$EDX_UI_TOOLKIT_VENDOR_PATH/js"
+cp --recursive \
+	"$NODE_MODULES_PATH/edx-ui-toolkit/src/js" \
+	"$EDX_UI_TOOLKIT_VENDOR_PATH"
+
+# Copy certain node_modules libraries into vendor directory.
+cp --force \
+	"$NODE_MODULES_PATH/backbone.paginator/lib/backbone.paginator.js" \
+	"$NODE_MODULES_PATH/backbone/backbone.js" \
+	"$NODE_MODULES_PATH/bootstrap/dist/js/bootstrap.bundle.js" \
+	"$NODE_MODULES_PATH/hls.js/dist/hls.js" \
+	"$NODE_MODULES_PATH/jquery-migrate/dist/jquery-migrate.js" \
+	"$NODE_MODULES_PATH/jquery.scrollto/jquery.scrollTo.js" \
+	"$NODE_MODULES_PATH/jquery/dist/jquery.js" \
+	"$NODE_MODULES_PATH/moment-timezone/builds/moment-timezone-with-data.js" \
+	"$NODE_MODULES_PATH/moment/min/moment-with-locales.js" \
+	"$NODE_MODULES_PATH/picturefill/dist/picturefill.js" \
+	"$NODE_MODULES_PATH/requirejs/require.js" \
+	"$NODE_MODULES_PATH/underscore.string/dist/underscore.string.js" \
+	"$NODE_MODULES_PATH/underscore/underscore.js" \
+	"$NODE_MODULES_PATH/which-country/index.js" \
```

Comment:
This `cp` command is not going to replicate the previous paver command. For instance: `$NODE_MODULES_PATH/which-country/index.js` will be moved to `"JS_VENDOR_PATH/index.js`. But it should be moved to `"JS_VENDOR_PATH/which-country/index.js`. (if I understand the code correctly)

Reply:
I think the code here is correct, although I wrote it months ago so I'm glad you are having me double-check.

If you run `tutor local run lms openedx-assets npm` on tutor master (which calls the old `process_npm_assets`), you get:
```
...
Copying vendor files into static directory
/bin/cp -rf node_modules/backbone.paginator/lib/backbone.paginator.js common/static/common/js/vendor
/bin/cp -rf node_modules/backbone/backbone.js common/static/common/js/vendor
/bin/cp -rf node_modules/bootstrap/dist/js/bootstrap.bundle.js common/static/common/js/vendor
... etc
Copying vendor library dir: node_modules/@edx/studio-frontend/dist/
/bin/cp -rf node_modules/@edx/studio-frontend/dist/assets.min.js.map common/static/common/js/vendor
/bin/cp -rf node_modules/@edx/studio-frontend/dist/courseHealthCheck.min.js common/static/common/js/vendor
/bin/cp -rf node_modules/@edx/studio-frontend/dist/common.min.js common/static/common/js/vendor
/bin/cp -rf node_modules/@edx/studio-frontend/dist/common.min.css common/static/common/css/vendor
/bin/cp -rf node_modules/@edx/studio-frontend/dist/courseHealthCheck.min.js.map common/static/common/js/vendor
... etc
...
```
It's copying the files directly into the `common/static/common/[js|css]/vendor` directories. This matches my reading of [copy_vendor_library](https://github.com/openedx/edx-platform/blob/open-release/olive.master/pavelib/assets.py#L607).

The corresponding output from this PR's `process-npm-assets.sh` script is:
```
+ find ./node_modules/@edx/studio-frontend/dist -type f ! -name *.css ! -name *.css.map -print0
+ xargs --null cp --target-directory=./common/static/common/js/vendor
+ find ./node_modules/@edx/studio-frontend/dist -type f ( -name *.css -o -name *.css.map+  ) -print0
xargs --null cp --target-directory=./common/static/common/css/vendor
+ rm -rf ./common/static/edx-ui-toolkit/js
+ cp --recursive ./node_modules/edx-ui-toolkit/src/js ./common/static/edx-ui-toolkit
+ cp --force ./node_modules/backbone.paginator/lib/backbone.paginator.js ./node_modules/backbone/backbone.js ./node_modules/bootstrap/dist/js/bootstrap.bundle.js ./node_modules/hls.js/dist/hls.js ./node_modules/jquery-migrate/dist/jquery-migrate.js ./node_modules/jquery.scrollto/jquery.scrollTo.js ./node_modules/jquery/dist/jquery.js ./node_modules/moment-timezone/builds/moment-timezone-with-data.js ./node_modules/moment/min/moment-with-locales.js ./node_modules/picturefill/dist/picturefill.js ./node_modules/requirejs/require.js ./node_modules/underscore.string/dist/underscore.string.js ./node_modules/underscore/underscore.js ./node_modules/which-country/index.js ./common/static/common/js/vendor
```
(The `find ... | xargs cp ...` calls correspond to the `Copying vendor library dir: node_modules/@edx/studio-frontend/dist/` section. The simple `cp` calls correspond to the `Copying vendor files into static directory` section.)